cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,39 @@
+/* eslint-disable */
 /// <reference types='cypress' />
 
+const { generateUser } = require('../support/generateUser')
 describe('Student Registration page', () => {
   before(() => {
+    cy.visit('https://demoqa.com/automation-practice-form')
+  })
 
-  });
+  it('should send the form with filled fields and check modal', () => {
+    const { userName, email, mobile, subject, currentAddress, state, city } =
+      generateUser()
 
-  it('', () => {
+    cy.findById('firstName').type(userName)
+    cy.findById('lastName').type(userName)
+    cy.findById('userEmail').type(email)
+    cy.findById('userNumber').type(mobile)
+    cy.findById('currentAddress').type(currentAddress)
+    cy.findById('state').type(state)
+    cy.findById('city').type(city)
+    cy.get('.subjects-auto-complete__placeholder').type(subject)
+    cy.get(`input[value='Female']`).check()
+    cy.get(`input[value='Female']`).should('be.checked')
+    cy.get(`input[id='hobbies-checkbox-1']`).check()
+    cy.get(`input[id='hobbies-checkbox-1']`).should('be.checked')
 
-  });
-});
+    cy.get(`button[id='submit']`).click()
```

Comment:
Consider adding an assertion to verify that the form submission was successful, such as checking for a confirmation message or modal.